aioredlock/redis.py

```diff
@@ -403,7 +413,12 @@ async def unset_lock(self, resource, lock_identifier):
 
         if not unlocked:
             errors = [e for e in successes if isinstance(e, BaseException)]
-            raise LockError('Can not release the lock') from errors[0]
+            if any(type(e) is LockRuntimeError for e in errors):
+                raise [e for e in errors if type(e) is LockRuntimeError][0]
+            elif any(type(e) is LockAcquiringError for e in errors):
+                raise [e for e in errors if type(e) is LockAcquiringError][0]
+            else:
+                raise LockError('Can not release the lock') from errors[0]
```

Comment:
Can we make this into a function, and pass in the errors, and also the message for LockError instead of having this same block of code everywhere?

thanks

Reply:
OK, just pushed a new version!